src/mlpack/methods/naive_bayes/naive_bayes_classifier.hpp

```diff
@@ -169,59 +182,51 @@ class NaiveBayesClassifier
    * @param probabilities This will be filled with class probabilities for each
    *      point. Each row represents a point.
    */
+  template<typename MatType>
   void Classify(const MatType& data,
                 arma::Row<size_t>& predictions,
-                arma::mat& probabilities) const;
+                ModelMatType& probabilities) const;
```

Comment:
I think that `ProbabilitiesVecType` should always have the same type as the underlying `ModelMatType`, but I think it is fine to templatize it.  I'll push changes soon.